### PR TITLE
drop type specs in RecoTauTag

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Update the safer syntax for existing parameter :
- drop type specifications where the original parameter exists.
- move all parameter inside the clone

Instead of modifying parameters with full type specs, which can be interpreted as an insertion of a new parameter, it is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.
(The previous PR were [PR#31162](https://github.com/cms-sw/cmssw/pull/31162),[PR#31243](https://github.com/cms-sw/cmssw/pull/31243),[PR#31332](https://github.com/cms-sw/cmssw/pull/31332), [PR#31389](https://github.com/cms-sw/cmssw/pull/31389), [PR#31523](https://github.com/cms-sw/cmssw/pull/31523), [PR#31538](https://github.com/cms-sw/cmssw/pull/31538))

In this PR, total 17 files changed. 
- RecoTauTag/Configuration  : 4 files 
- RecoTauTag/HLTProducers : 2 files
- RecoTauTag/RecoTau :  11 files 

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_11_2_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoTauTag/Configuration/python/HPSPFTaus_cff.py [hunk 1]
```diff
@@ -423,11 +423,11 @@
     verbosity = 0
```
Comment (perrotta) [inline RecoTauTag/Configuration/python/HPSPFTaus_cff.py:423]: at line 422 there is also  a forgotten
```PYTHON
    inputIDNameSuffix = cms.string("dR03"),
```

File: RecoTauTag/Configuration/python/boostedHPSPFTaus_cff.py [hunk 2]
```diff
@@ -23,11 +22,11 @@
 ca8PFJetsCHSprunedForBoostedTaus = boostedTaus2.ak4PFJets.clone(
     boostedTaus3.CMSBoostedTauSeedingParameters,
     #src = cms.InputTag('pfNoPileUpForBoostedTaus'),
```
Comment (perrotta) [inline RecoTauTag/Configuration/python/boostedHPSPFTaus_cff.py:24]: even if commented out, you can remove type specification also here

File: RecoTauTag/RecoTau/python/RecoTauPiZeroBuilderPlugins_cfi.py [hunk 3]
```diff
@@ -68,7 +68,7 @@
 # and eta x phi size of strip increasing for low pT photons
```
Comment (perrotta) [inline RecoTauTag/RecoTau/python/RecoTauPiZeroBuilderPlugins_cfi.py:69]: `plugin` parameter in `modStrips` forgotten

File: RecoTauTag/RecoTau/python/RecoTauPiZeroBuilderPlugins_cfi.py [hunk 4]
```diff
@@ -68,7 +68,7 @@
...
 
 modStrips2 = strips.clone(
```
Comment (perrotta) [inline RecoTauTag/RecoTau/python/RecoTauPiZeroBuilderPlugins_cfi.py:70]: Also forgotten `makeCombinatoricStrips` at L84

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Type specification dropped from cloned and modified configs as planned
- Jenkins tests pass and show no differences wrt baseline

**Comment by Martin-Grunewald** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**